packages/react-reconciler/src/__tests__/ReactIncrementalUpdates-test.js

```diff
@@ -196,11 +196,13 @@ describe('ReactIncrementalUpdates', () => {
       // Now flush the remaining work. Even though e and f were already processed,
       // they should be processed again, to ensure that the terminal state
       // is deterministic.
-      // TODO: should d, e, f be flushed again first?
       expect(Scheduler).toFlushAndYield([
+        // Since 'g' is in a transition, we'll process 'd' separately first.
```

Comment:
I think one thing that makes this test confusing is that there are both "default" sync and "discrete sync" updates. Since this test is about interruptibility/rebasing, I would remove the `gate` and instead wrap every update in either `startTransition` or `flushSync`.